tools/cli.js

```diff
@@ -247,12 +268,19 @@ lineProcessor.event = (tokens, callback) => {
 };
 
 lineProcessor.connect = (tokens, callback) => {
-  if (tokens === undefined || tokens.trim().startsWith(':')) {
+  if (tokens === undefined) {
     return callback(reportMissingArgument('Host'));
   }
   const args = _split(tokens, ' ', 2);
-  const [host, portString] = _split(args[0], ':');
-  if (portString === undefined) {
+  let [scheme, authority] = _split(args[0], '://', 1, true);
```

Comment:
Tried it at first, but it turned out to be difficult to reply with meaningful error messages when using `url.parse()`.